lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -778,27 +790,67 @@ instance Buildable (Block) where
         <> build h
         <> if null txs then " ∅" else "\n" <> indentF 4 (blockListF txs)
 
+data BlockHeader = BlockHeader
+    { slotNo
+        :: SlotNo
+    , blockHeight
+        :: Quantity "block" Word32
+    , headerHash
+        :: !(Hash "BlockHeader")
+    , parentHeaderHash
+        :: !(Hash "BlockHeader")
+    } deriving (Show, Eq, Ord, Generic)
+
+-- | Magic value denoting the hash of a block that \"does not exist\".
+-- Specifically, this is the hash of \"the parent block\" of the genesis block.
+hashOfNoParent :: Hash "BlockHeader"
```

Comment:
another approach could be to use introduce something like "parent" with  Maybe...and in that case no need for magic values...it would be just Nothing

Reply:
Hm. 🤔  I agree — but unfortunately, the database does serialize the `parentHeaderHash` in the table:

https://github.com/input-output-hk/cardano-wallet/blob/3a24c14b3208da7a6e890de1c2f3f261ab8d5b01/lib/core/src/Cardano/Wallet/DB/Sqlite/TH.hs#L210

and then we would have the magic value in the serialization. But I suppose that is an improvement over having the magic value in the type.

Reply:
yes, we could probably deal with this magic value at the level of `PersistField` instance or so. and everywhere above we could have Maybe. just suggestion

Reply:
so have magic value defined somewhere here https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Wallet/DB/Sqlite/Types.hs and `toPersistValue Nothing = magicValue`

Reply:
I have added the `hashOfNoParent` constant to this module, but stopped short of changing the `BlockId` type. Instead, I have created two conversion functions.